src/mlpack/methods/decision_tree/gini_gain.hpp

```diff
@@ -88,18 +90,28 @@ class GiniGain
 
       // SIMD loop: add counts for four elements simultaneously (if the compiler
       // manages to vectorize the loop).
+      #pragma omp parallel for reduction (+:accWeights)
```

Comment:
Hi, The first error is associated with reduction, I think it must be a scalar but here it is an array of type double. Also 

> OpenMP predefines a set of reduction operators. Each reduction variable must be a scalar (for example, int, long, and float).

So I don't think a double type will also work.
 For more info, refer this [link](https://docs.microsoft.com/en-us/cpp/parallel/concrt/convert-an-openmp-loop-that-uses-a-reduction-variable?view=vs-2019).

Reply:
Hello, thanks for reviewing my work.Ill make all the changes push it up again for review.